### DISCUSSION TITLE
Webp to jpg

### DISCUSSION THREAD

**Discussion by crownaintanoob** [xulihang/ImageTrans-docs#342] [Show and tell]
**Since webp files don't work with ImageTrans, I fixed it using Node.js.**
_**Download Node.js from https://nodejs.org/en/download/**_

**Here are the files needed for the fix, and where you need to put them:**
1. Make a folder called imagesDownloads in the ImageTrans folder
2. Download this and put the folder that is inside the zip file into the ImageTrans Folder: [node_modules.zip](https://github.com/xulihang/ImageTrans-docs/files/10329175/node_modules.zip)
3.  Download this and put the javascript file that is inside the zip file into the ImageTrans Folder: [ImageFixUploader.zip](https://github.com/xulihang/ImageTrans-docs/files/10332491/ImageFixUploader.zip)
4. Download this and put the bat file that is inside the zip file into the ImageTrans folder: [ServerImageFixUploader.zip](https://github.com/xulihang/ImageTrans-docs/files/10329216/ServerImageFixUploader.zip)
5. Download the zip file, unzip it and install it ([How to install unpacked extensions in chrome](https://www.youtube.com/watch?v=oswjtLwCUqg)): [ImageTransExtension.zip](https://github.com/xulihang/ImageTrans-docs/files/10329220/ImageTransExtension.zip)

**How to use the fix:**

1. Open ImageTrans, start the server: 
![image](https://user-images.githubusercontent.com/70343800/210180642-278e9744-3925-4e51-9c03-55b974d320da.png)
2. Open the file called ServerImageFixUploader.bat in the ImageTrans folder, keep it open as it will fix the webp files.
3. Now use the custom extension, it should fix the webp files when using the server translation on websites that use webp files.

**Here is a website that uses webp files, so that you can try the fix:**
https://www.webmota.com

_**Let me know what you think of my fix!**_

**Comment by xulihang** [comment]
Good workground and the modified extension with auto translate all images feature.

**Reply by crownaintanoob** [reply]
> Good workground and the modified extension with auto translate all images feature.

Thanks!
